tests/src/test/scala/com/nvidia/spark/rapids/OrcQuerySuite.scala

```diff
@@ -92,4 +97,53 @@ class OrcQuerySuite extends SparkQueryCompareTestSuite {
   ) {
     frame => frame
   }
+
+  private def getColumnEncoding(orcDir: File): (String, String) = {
+    val orcFile = orcDir.listFiles(f => f.getName.endsWith(".orc"))(0)
+    val p = new Path(orcFile.getCanonicalPath)
+    val conf = OrcFile.readerOptions(new Configuration())
+    var reader: Reader = null
+    try {
+      reader = OrcFile.createReader(p, conf)
+      withResource(reader.rows().asInstanceOf[RecordReaderImpl]) { rows =>
+        val stripe: StripeInformation = reader.getStripes.get(0)
+        val stripeFooter = rows.readStripeFooter(stripe)
+        (stripeFooter.getColumns(1).getKind.toString,
+            stripeFooter.getColumns(2).getKind.toString)
+      }
+    } finally {
+      reader match {
+        case c: AutoCloseable => c.close()
+        case _ => // cdh321, cdh330 use a lower ORC version, and the reader is not a AutoCloseable
+      }
+    }
+  }
+
+  /**
+   * This corresponds to Spark case:
+   * https://github.com/apache/spark/blob/v3.4.0/sql/core/src/test/scala/org/apache/spark/sql/
+   * execution/datasources/orc/OrcQuerySuite.scala#L359
+   */
+  test("SPARK-5309 strings stored using dictionary compression in orc") {
+    withGpuSparkSession(spark => {
```

Comment:
Would it be better to have the CPU write one file and the GPU write another file, and then we can do a comparison of the footer to verify specific things about the encoding selected and the effectiveness of the encoding?  This can be a follow on issue. I just think that we are missing out on a lot of potential here.

Reply:
Done. Now compare the CPU and GPU directory encoding.